### PR TITLE
Support new dangerzone endpoint access from weave-trace server

### PR DESCRIPTION
**Pull request by KyleGoyette**
In:  https://github.com/wandb/core/pull/25229 we added a new endpoint, to be used internally to pass around sensitive information. This endpoint relies on authentication through JWTs, to allow other services in wandb to fetch customer secret values.

This PR:
1. Creates a new oidc reviewer cluster role binding that allows unauthenticated access to get the oidc configuration
2. Adds an environment variable `GORILLA_INTERNAL_JWT_SUBJECTS_TO_ISSUERS` to the app chart for gorilla to set up the issuer and subject relation.

This unlocks the weave playground which relies on fetching user secrets into the weave-trace service to make requests to llm services.

note: their is a PR in core (https://github.com/wandb/core/pull/25352) to make use of the environment variable structure used here.

### CODE REVIEW

File: charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl [hunk 1]
```diff
@@ -4,19 +4,19 @@
...
+{{- default "weave-trace" .Values.nameOverride | trunc 63 | trimSuffix "-" }}
 {{- end }}
 
 {{/*
 Create a default fully qualified name for weave-trace. (Should be something like wandb-weave-trace)
 We truncate at 63 chars because some Kubernetes name fields are limited to this (by the DNS naming spec).
 If release name contains chart name it will be used as a full name.
 */}}
-{{- define "weaveTrace.fullname" -}}
+{{- define "weave-trace.weaveTrace.fullname" -}}
```
Comment (jsbroks) [inline charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl:15]: we shouldn't be including weave-traces infront of everything, thats inherited by helm

File: charts/operator-wandb/templates/oidc.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (jsbroks) [inline charts/operator-wandb/templates/oidc.yaml:1]: This should all be properly templated out using chart prefix etc.

File: charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl [hunk 3]
```diff
@@ -4,19 +4,19 @@
...
 {{- if .Values.fullnameOverride }}
 {{- .Values.fullnameOverride | trunc 63 | trimSuffix "-" }}
 {{- else }}
-{{- $name := default .Chart.Name .Values.nameOverride }}
+{{- $name := default "weave-trace" .Values.nameOverride }}
```
Comment (jsbroks) [inline charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl:19]: why are we hard coding this?

File: charts/operator-wandb/charts/weave-trace/templates/serviceaccount.yaml [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+    {{- $weaveLabels := index .Values.global "weave-trace" "serviceAccount" "labels" }}
+    {{- if $weaveLabels -}}
+    {{-   toYaml $weaveLabels | nindent 4 }}
     {{- end }}
   annotations:
-    {{- if .Values.serviceAccount.annotations -}}
-    {{-   toYaml .Values.serviceAccount.annotations | nindent 4 }}
+    {{- $weaveAnnotations := index .Values.global "weave-trace" "serviceAccount" "annotations" }}
+    {{- if $weaveAnnotations -}}
+    {{-   toYaml $weaveAnnotations | nindent 4 }}
```
Comment (jsbroks) [inline charts/operator-wandb/charts/weave-trace/templates/serviceaccount.yaml:16]: Why are we changing these patterns?

File: charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl (hunk 3 above)
Comment (KyleGoyette) [reply charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl:19]: This function is now called in the app server context so it was using the wrong name when called there. There's probably a better way for me to do it.

Comment (jsbroks) [reply charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl:19]: Whats do you mean by app server context?

Comment (danielpanzella) [reply charts/operator-wandb/charts/weave-trace/templates/_helpers.tpl:19]: technically, you should not be calling a template across charts

File: charts/operator-wandb/charts/app/templates/_deployment.tpl [hunk 5]
```diff
@@ -260,6 +260,10 @@ spec:
               value: {{ .Values.artifactsGc.NumWorkers | quote }}
             - name: GORILLA_ARTIFACTS_GC_DELETE_FILES_NUM_WORKERS
               value: {{ .Values.artifactsGc.DeleteFilesNumWorkers | quote }}
+            {{- if index .Values.global "weave-trace" "enabled" }}
+            - name: GORILLA_INTERNAL_JWT_SUBJECTS_TO_ISSUERS
+              value: '{"system:serviceaccount:default:{{ include "weave-trace.weaveTrace.serviceAccountName" . }}": "https://kubernetes.default.svc.cluster.local"}'
+            {{- end }}
```
Comment (jsbroks) [inline charts/operator-wandb/charts/app/templates/_deployment.tpl:266]: Something seems wrong here. We also cannot do direct namespace references

**Review by danielpanzella — APPROVED** [review]

**Pull request merged**